sections/Claim/ConvertSection/ConvertSection.tsx

```diff
@@ -0,0 +1,296 @@
+import { DEFAULT_NETWORK_ID } from 'constants/defaults';
+import { TransactionStatus } from 'constants/transactions';
+
+import { getKeyValue } from 'utils/helpers';
+import { getGasEstimateWithBuffer } from 'utils/network';
+
+const ConvertSection = () => {
+	const [hasAllowance, setHasAllowance] = useState<boolean>(false);
+	const [hasVAelinBalance, setHasVAelinBalance] = useState<boolean>(false);
+	const [showConverterTxModal, setConverterTxModal] = useState<boolean>(false);
+	const [gasvAelinConverterLimitEstimate, setvAelinConverterGasLimitEstimate] =
+		useState<GasLimitEstimate>(null);
+
+	const { gasPrice, setGasPrice, setTxState } = TransactionData.useContainer();
+	const { monitorTransaction } = TransactionNotifier.useContainer();
+	const { walletAddress, network, signer } = Connector.useContainer();
+
+	const vAelinConverterContract = useMemo(() => {
+		if (!signer || !network?.id) return null;
+
+		const vAelinConverterContract = (getKeyValue(VAelinConverterContract) as any)(
+			network?.id ?? DEFAULT_NETWORK_ID
+		);
+
+		if (!vAelinConverterContract) return null;
+
+		return new ethers.Contract(
+			vAelinConverterContract.address,
+			vAelinConverterContract.abi,
+			signer
+		);
+	}, [signer, network?.id]);
+
+	const vAelinTokenContract = useMemo(() => {
+		if (!signer || !network?.id) return null;
+
+		const vAelinTokenContract = (getKeyValue(VAelinTokenContract) as any)(
+			network?.id ?? DEFAULT_NETWORK_ID
+		);
+
+		if (!vAelinTokenContract) return null;
+
+		return new ethers.Contract(vAelinTokenContract.address, vAelinTokenContract.abi, signer);
+	}, [signer, network?.id]);
+
+	useEffect(() => {
+		const getGasLimitEstimate = async () => {
+			if (!walletAddress || !vAelinConverterContract || !vAelinTokenContract) return;
+			try {
+				setvAelinConverterGasLimitEstimate(null);
+				if (!hasAllowance) {
+					const gasLimit = await vAelinTokenContract.estimateGas.approve(
+						vAelinConverterContract.address,
+						ethers.constants.MaxUint256
+					);
+					setvAelinConverterGasLimitEstimate(wei(gasLimit, 0));
+				} else {
+					const gasEstimate = await vAelinConverterContract.estimateGas.convertAll();
+					setvAelinConverterGasLimitEstimate(wei(gasEstimate, 0));
+				}
+			} catch (e) {
+				console.log(e);
+				setvAelinConverterGasLimitEstimate(null);
+			}
+		};
+		getGasLimitEstimate();
+	}, [walletAddress, network.id, vAelinConverterContract, vAelinTokenContract, hasAllowance]);
+
+	const getAllowance = useCallback(async () => {
+		if (!vAelinConverterContract || !vAelinTokenContract || !walletAddress) return;
+		try {
+			const allowance = await vAelinTokenContract.allowance(
+				walletAddress,
+				vAelinConverterContract.address
+			);
+			setHasAllowance(!!Number(ethers.utils.formatEther(allowance)));
+		} catch (e) {
+			console.log(e);
+			setHasAllowance(false);
+		}
+	}, [vAelinTokenContract, vAelinConverterContract, walletAddress]);
+
+	const getVAelinBalance = useCallback(async () => {
```

Comment:
Can you please have a look at what I did for the staking contract? You can use a query I did for this. `useGetTokenBalance`. All you have to do is pass a contract address to the query. Please do not reinvent the wheel and make sure to reuse all the existing components/queries if possible.

Reply:
Done